numpy/core/_add_newdocs.py

```diff
@@ -7970,62 +7970,169 @@ def luf(lamdaexpr, *args, **kwargs):
 #
 ##############################################################################
 
-add_newdoc('numpy.core.numerictypes', 'bool_',
-    """NumPy's Boolean type.  Character code: ``?``.  Alias: bool8""")
+def get_type(place, obj):
```

Comment:
This `place` argument seems kind of redundant. Perhaps just add `import numpy.core.numerictypes as _types` outside these functions, and then change this function to `getattr(_types, obj, None)`

Reply:
Looks good, but the import fails when I do that:
```
>>> import numpy as np
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/yannick/numpy/env/lib/python3.6/site-packages/numpy-1.16.0.dev0+2c80d46-py3.6-linux-x86_64.egg/numpy/__init__.py", line 142, in <module>
    from . import core
  File "/home/yannick/numpy/env/lib/python3.6/site-packages/numpy-1.16.0.dev0+2c80d46-py3.6-linux-x86_64.egg/numpy/core/__init__.py", line 64, in <module>
    from . import _add_newdocs
  File "/home/yannick/numpy/env/lib/python3.6/site-packages/numpy-1.16.0.dev0+2c80d46-py3.6-linux-x86_64.egg/numpy/core/_add_newdocs.py", line 7973, in <module>
    import numpy.core.numerictypes as _types
AttributeError: module 'numpy' has no attribute 'core' 
```

Reply:
The trick's stolen from `add_newdoc` in `numpy/core/function_base.py`, btw: https://github.com/numpy/numpy/blob/master/numpy/core/function_base.py#L366 It's semi-hocuspocus, to me, but I'm assuming that since the `_add_newdocs.py` is run while importing, the submodule is not completely there yet to be imported in the normal way?

Reply:
`from numpy.core import numerictypes as _types` does work, so I can use that.